js/data/bucket.js

```diff
@@ -122,17 +123,17 @@ Bucket.prototype.populateBuffers = function() {
  * `vertexLength` vertices. If not, append a new elementGroup. Should be called
  * by `populateBuffers` and its callees.
  * @private
- * @param {string} programName the name of the program associated with the buffer that will receive the vertices
+ * @param {string} dataLayerTypeName type of buffer that will receive the vertices
```

Comment:
All good points. I feel less good about these names in hindsight. I will revert as much renaming as possible within this PR.